### DISCUSSION TITLE
Deamon/Service or docker to extract metadata from files

### DISCUSSION THREAD

**Discussion by SteffenBrinckmann** [TheELNConsortium/TheELNFileFormat#47] [Ideas]
I guess we also should talk about a common format on how to extract metadata from files. 
The scientist has a collection of extractors for different file-formats, since these extractors follow the same structure, they can be exchanged between scientists,ELNs.... They are basically small scripts that get a filename and extract metadata (vendor specific, thumbnail, perhaps scientist adds metadata (e.g. average pixel intensity))
The to-be created software layer identifies the correct extractor and uses it to extract the metadata. One route would be to create a docker/virtualization that receives the information, does the trick and returns the metadata. Other route would be to create a deamon (Linux/Mac) and Service (Win), that basically does the same. 
What do you think?

**Comment by NicolasCARPi** [comment]
I think it should be a web service that could run along side other software, to fit with the "micro-service" trend. That would be a cool project to start!

**Comment by SteffenBrinckmann** [comment]
OK, I started with a basic version, see repo. I hope everything is clean and perfect. It just defines the interfaces and then one can implement service or docker doing these things. Nico, don't rip my fantasy too much into shreds. ;-)

**Comment by ml-evs** [comment]
Hi, just stumbled across this discussion and the other repo, I think we already tried to reach out (about the ELN API work) but we are putting together a proposal for a MaRDA working group along the same lines as this, would be great to join forces! There has been some delay getting started, but let me know if you would like to be involved. You can see the proposal here: https://docs.google.com/document/d/1_D3X1tQGqGPjktH8WpzffzfdaBjgvB3JjwaVJ_sTg9E/edit?usp=sharing (if you want to be involved just request edit access and add your names!)

**Reply by NicolasCARPi** [reply]
request sent.
